jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildSteps.java

```diff
@@ -55,19 +55,19 @@ public static BuildSteps forBuildToDockerRegistry(BuildConfiguration buildConfig
         DESCRIPTION_FOR_DOCKER_REGISTRY,
         buildConfiguration,
         () ->
```

Comment:
Looks like the sole reason we were accepting a lambda here was not to run `StepsRunner` right away. With the change to `StepsRunner` in this PR, just instantiating `StepsRunner` with `.begin()` won't run the steps right away (if we did not call the `wait...Image()` methods.) Previously, `BuildSteps.run()` would kick the steps. So, what if we remove the lambda wrapping that now seems unnecessary? That is,

1. Revert `wait...Image()` and remove the call to `runStepsRunners()`.
1. Change the `BuildSteps` constructor to accept a `StepsRunner` instance, instead of a lambda wrapping that.
1. Change `BuildSteps.run()` to call `StepsRunner.runStepsRunner()` there.

Haven't really tested out this, but I think this will work?

Reply:
Yep that would work! Will change it to that